NEXTCLOUD.cmake

```diff
@@ -15,6 +15,30 @@ set( APPLICATION_VIRTUALFILE_SUFFIX "nextcloud" CACHE STRING "Virtual file suffi
 set( LINUX_PACKAGE_SHORTNAME "nextcloud" )
 set( LINUX_APPLICATION_ID "${APPLICATION_REV_DOMAIN}.${LINUX_PACKAGE_SHORTNAME}")
 
+option(DEV_BUILD "If enabled, name variables will be appended with safe variations of DEV_SUFFIX in order to reduce conflicts." OFF)
+
+if (NOT DEV_SUFFIX)
+    set(DEV_SUFFIX "Dev"
+        CACHE STRING
+        "Choose a suffix with which to append build name variables when DEV_BUILD is enable in order to prevent conflicts.
+        Spaces and parentheses will be stripped out where necessary. Other symbols may not be safe."
+        FORCE)
+endif ()
+
+if (DEV_BUILD)
+    message(STATUS "DEV_BUILD is set to ${DEV_BUILD}, and name variables will be appended with safe variations of DEV_SUFFIX value \"${DEV_SUFFIX}\" to prevent conflicts.")
+    string(REPLACE " " "" SUFFIX_STRIPPED "${DEV_SUFFIX}")
+    string(REPLACE "(" "" SUFFIX_STRIPPED "${SUFFIX_STRIPPED}")
+    string(REPLACE ")" "" SUFFIX_STRIPPED "${SUFFIX_STRIPPED}")
```

Comment:
That's not needed anymore

Reply:
It's there because I personally want to be able to use spaces and parentheses. Just because it isn't in the default string doesn't mean it doesn't do anything.

Reply:
But then whats with other special signs? Should we handle them also? For me it is a little bit inconsistent to just handle spaces and parens. It introduces a lot of noise and is a source for potential bugs. At the moment we don't need that functionality and if we feel like we need it, we can add it at any time:)

Reply:
https://cmake.org/cmake/help/latest/policy/CMP0037.html

`string(REGEX REPLACE [^a-zA-Z0-9\._+-] "" SUFFIX_STRIPPED "${DEV_SUFFIX}")`